src/metabase/task.clj

```diff
@@ -210,6 +210,12 @@
   (when-let [scheduler (scheduler)]
     (qs/delete-trigger scheduler trigger-key)))
 
+(mu/defn delete-all-triggers-of-job!
+  "Delete all triggers for a given job key."
+  [job-key :- (ms/InstanceOfClass JobKey)]
+  (when-let [scheduler (scheduler)]
+    (qs/delete-triggers scheduler (map #(.getKey ^Trigger %) (qs/get-triggers-of-job scheduler job-key)))))
```

Comment:
Since this ends up enumerating all the triggers of the job anyway, why not take a predicate we can use to filter out the ones that *should* exist?

I'm nervous about missing triggers in the gap between deleting triggers and re-creating them.